tools/python/src/simple_object_detector_py.h

```diff
@@ -12,7 +12,8 @@
 namespace dlib
 {
     typedef object_detector<scan_fhog_pyramid<pyramid_down<6> > > simple_object_detector;
-
+    typedef scan_fhog_pyramid<pyramid_down<6> > image_scanner_type;
```

Comment:
I think you should be able to use the simple_object_detector everywhere here, so this new typedef isn't needed.  It would also make it clear that all these pieces of code must operate on the same type, simple_object_detector, which I'm pretty sure is the case.